spec/abstract/KHeap_A.thy

```diff
@@ -223,9 +223,20 @@ abbreviation
   set_reply :: "obj_ref \<Rightarrow> reply \<Rightarrow> (unit,'z::state_ext) s_monad" where
   "set_reply \<equiv> set_simple_ko Reply"
 
+definition
+  update_reply :: "obj_ref \<Rightarrow> (reply \<Rightarrow> reply) \<Rightarrow> (unit,'z::state_ext) s_monad"
+where
+  "update_reply ptr f  \<equiv> do
+     obj \<leftarrow> get_object ptr;
+     case obj of Reply reply \<Rightarrow> set_object ptr (Reply (f reply)) | _ \<Rightarrow> fail
```

Comment:
I haven't looked into the details in a while, but did we want to use `set_object`, or do we prefer `set_simple_ko`? I was thinking of making a similar "update" function for the endpoints and notifications, so maybe we should decide on the right pattern for this.